test/test_images/readiness/readiness.go

```diff
@@ -0,0 +1,73 @@
+/*
+Copyright 2021 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"flag"
+	"fmt"
+	"log"
+	"net/http"
+	"os"
+	"sync"
+	"time"
+
+	"knative.dev/serving/test"
+)
+
+func main() {
+	// Exec probe.
+	flag.Parse()
+	args := flag.Args()
+	if len(args) > 0 && args[0] == "probe" {
+		if _, err := http.Get(os.ExpandEnv("http://localhost:$PORT/")); err != nil {
+			log.Fatal("Failed to probe ", err)
+		}
+		return
+	}
+
+	// HTTP Probe.
+	healthy := true
+	var mu sync.Mutex
```

Comment:
Total nit: Should we use an atomic boolean here?

Reply:
we certainly could; I'm intending this image to get extended a bit to eg bind a tcp listener on a different port to properly test tcp probes and to go unready and ready so seems like the mutex might end up being useful, but possibly atomic bool would actually work fine for most of those 🤷‍♂️. TBH the mutex for some reason reads slightly clearer to me here, I think, but if you feel even vaguely strongly I'll swap it :)